### ISSUE TITLE
Deprecated API incorrectly shown

### ISSUE THREAD

**Issue by vnesterova-lohika-tix** [doitintl/kube-no-trouble#309]
Hi. When I run kubent for my cluster, I get results that some API deprecated, for example:
`Ingress   kubecost   kubecost  networking.k8s.io/v1beta1  networking.k8s.io/v1`
but when I open Ingress itself, I see this:
```
apiVersion: networking.k8s.io/v1
kind: Ingress
metadata:
  name: kubecost
  namespace: kubecost
  managedFields:
    - manager: Go-http-client
      operation: Update
      apiVersion: networking.k8s.io/v1beta1
      fieldsType: FieldsV1
```
I suppose that kubent is looking into "managedFields" part as well and shows deprecated API from there. But according to documentation, this is a server-side apply Kubernetes feature that should not be manually edited by user.

**Comment by Tomasz-Kluczkowski** [comment]
I also noticed this for example for StorageClass defined like shown below.
We can see that the resource is already at newest api version `storage.k8s.io/v1`.
I am not certain but either it looks in managed fields or last applied configuration and reports it as outdated, when that is really not the case.

This breaks the usefulness of this wonderful tool to us as a way of automatically report outdated apis in our k8s clusters.

Would it be possible to either stop doing this false reports of outdated resource existing or add a flag that would skip checking managed fields / last applied configuration ?

```yaml
allowVolumeExpansion: true
apiVersion: storage.k8s.io/v1
kind: StorageClass
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"allowVolumeExpansion":true,"apiVersion":"storage.k8s.io/v1beta1","kind":"StorageClass","metadata":{"annotations":{"storageclass.beta.kubernetes.io/is-default-class":"true"},"labels":{"kubernetes.io/cluster-service":"true"},"name":"default"},"parameters":{"cachingmode":"ReadOnly","kind":"Managed","storageaccounttype":"StandardSSD_LRS"},"provisioner":"kubernetes.io/azure-disk"}
    storageclass.beta.kubernetes.io/is-default-class: "true"
  creationTimestamp: "2020-09-11T17:08:55Z"
  labels:
    kubernetes.io/cluster-service: "true"
  managedFields:
  - apiVersion: storage.k8s.io/v1beta1.  <<<< -- is this causing the false report? We should not be editing those fields so why reporting on them?
    fieldsType: FieldsV1
    fieldsV1:
      f:allowVolumeExpansion: {}
      f:metadata:
        f:annotations:
...
...
...


```

Thx a bunch for looking into this

Tom

**Comment by ayush-jain1** [comment]
> these are "false alarms" really make it hard to properly evaluate the output of kubent ;(

+1 :(

**Comment by openstack-test** [comment]
Kubent output is inaccurate   +1

**Comment by stepanstipl** [comment]
Hi everyone,

so kubent does look **only** at the `last-applied-configuration` (or Helm in-cluster manifests if you use Helm). This is described in [README](https://github.com/doitintl/kube-no-trouble) and I've tried to explain it in https://github.com/doitintl/kube-no-trouble/issues/276#issuecomment-1062082004. This is due to how K8s work, and basically the original version used to create the resource is not preserved anywhere else (leaving aside newer managed fields for now).

As to `kubectl` - K8S will return the resource in the API version that was requested by the client, please see https://github.com/doitintl/kube-no-trouble/issues/276#issuecomment-1062928787. This version is independent of a) the version used when the resource was created and b) of the internal stored version.

**TL;DR:** If the `kubectl.kubernetes.io/last-applied-configuration` annotation contains old API version, it's expected behaviour this will be returned by kubent. This is how kubent works and I would say it's desirable, as it suggests the resource was created using the old API version, i.e. you will run into trouble if you try to do the same after cluster upgrade.

I can see that all the examples shared here contain the old API version in the annotation, so I'm going to close this issue. But if that's not the case, please comment. Thanks 👍
